src/containers/subjects-search/SubjectsSearch.jsx

```diff
@@ -0,0 +1,152 @@
+import ArrowBackIcon from '@mui/icons-material/ArrowBack'
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward'
+import SearchIcon from '@mui/icons-material/Search'
+import { Box, InputAdornment, TextField, Typography } from '@mui/material'
+import { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Link, useSearchParams } from 'react-router-dom'
+import AppButton from '~/components/app-button/AppButton'
+import AsyncAutocomplete from '~/components/async-autocomlete/AsyncAutocomplete'
+import TitleWithDescription from '~/components/title-with-description/TitleWithDescription'
+import { styles } from '~/containers/subjects-search/SubjectsSearch.styles'
+import useBreakpoints from '~/hooks/use-breakpoints'
+import { authRoutes } from '~/router/constants/authRoutes'
+import { categoryService } from '~/services/category-service'
+
+const SubjectsSearch = () => {
+  const { t } = useTranslation()
+  const { isMobile } = useBreakpoints()
+  const [searchParams, setSearchParams] = useSearchParams()
+  const [subjectName, setSubjectName] = useState(
+    searchParams.get('subjectName') || ''
+  )
+
+  const categoryName = searchParams.get('categoryName')
+
+  const handleCategoryChange = (_e, categoryValue) => {
+    if (!categoryValue) {
+      setSearchParams((params) => {
+        params.delete('categoryId')
+        params.delete('categoryName')
+        return params
+      })
+      return
+    }
+    setSearchParams((params) => {
+      params.set('categoryId', categoryValue._id)
+      params.set('categoryName', categoryValue.name)
```

Comment:
Why do we need both ID and name in the URL?

Reply:
I use categoryName to set the block title. If it is better to save it in the component state, I can try to fix that

![image](https://github.com/Space2Study-UA-1156/Client-01/assets/32570823/bcf1b711-b1aa-499e-a536-47c1de860965)